Atomic/syscontainers.py

```diff
@@ -1199,6 +1165,70 @@ def _do_checkout(self, repo, options):
 
         return options["values"]
 
+    def checkout_container_wrapup(self, rpm_install_content, tmpfiles_template, options, was_service_active):
+        """
+        Last stage of checking out a container. It includes outputting results,
+        setup for systemd part, and service starting
+        """
+        # Handle results based on rpm_install_content collected earlier
+        if rpm_install_content["rpm_file"]:
+            RPMHostInstall.install_rpm(rpm_install_content["rpm_file"])
+        else:
+            for installed_file in rpm_install_content["new_installed_files"]:
+                util.write_out("Created file {}".format(installed_file))
+
+        self._systemctl_command("daemon-reload")
+        if (tmpfiles_template):
+            self._systemd_tmpfiles("--create", options["tmpfilesout"])
+
+        # Note, when a rollback for container happens, the values from options will be reset to empty
+        self.checked_out_container_systemd_setup(options, was_service_active)
+
+    def checked_out_container_systemd_setup(self, options, was_service_active):
+        """
+        Handle container service accordingly based on was_service_active and cotnainer name.
+        If the container service was active before, the new container service will be started as well
+        """
+        if options["upgrade_mode"] == SystemContainers.CHECKOUT_MODE_INSTALL:
+            self._systemctl_command("enable", options["name"])
+        elif was_service_active:
+            if options["upgrade_mode"] == SystemContainers.CHECKOUT_MODE_UPGRADE_CONTROLLED:
+                self.start_container_and_rollback_if_necessary(options)
+            else:
+                self._systemctl_command("start", options["name"])
+
+    def start_container_and_rollback_if_necessary(self, options):
+        # Note, here we check if the container needs rollback by starting the
+        # container service. If no rollback is needed, the container service
+        # remains active
+        if(self.is_container_rollback_needed(options["name"])):
+            self.rollback_checked_out_container(options["name"])
+            options["values"] = {}
+
+    def is_container_rollback_needed(self, name):
+        try:
+            self._systemctl_command("start", name)
+        except subprocess.CalledProcessError:
+            return True
+        return False
+
+    def rollback_checked_out_container(self, name):
```

Comment:
same here, it can be part of the `start_container_and_rollback_if_necessary` function